Libraries/process/Sources/DefaultLauncher.cpp

```diff
@@ -55,60 +107,12 @@ ext::optional<int> DefaultLauncher::
 launch(Filesystem *filesystem, Context const *context)
 {
 #if _WIN32
-    WideString arguments;
-    for (std::string const &argument : context->commandLineArguments()) {
-        if (&argument != &context->commandLineArguments().front()) {
-            arguments += static_cast<wchar_t>(' ');
-        }
-
-        if (!argument.empty() && argument.find_first_of(" \t\n\v\"") == std::string::npos) {
-            arguments += StringToWideString(argument);
-        } else {
-            arguments += static_cast<wchar_t>('"');
-
-            WideString wide = StringToWideString(argument);
-            WideString special = StringToWideString("\"\\");
-            for (auto it = wide.begin(); it != wide.end(); ++it) {
-                /* Find the first backslash or quote. */
-                size_t first_ = wide.find_first_of(special, it - wide.begin());
-                auto first = (first_ == std::string::npos ? wide.end() : wide.begin() + first_);
-
-                /* Copy up to the first special character. */
-                arguments.insert(arguments.end(), it, first);
-
-                if (first == wide.end()) {
-                    /* Nothing special; jump to the end. */
-                    it = std::prev(first);
-                } else if (*first == static_cast<wchar_t>('"')) {
-                    /* Escape and append the quote itself. */
-                    arguments += static_cast<wchar_t>('\\');
-                    arguments += *first;
-
-                    /* Move past the quote. */
-                    it = first;
-                } else if (*first == static_cast<wchar_t>('\\')) {
-                    /* Find next non-backslash character after the first backslash. */
-                    size_t after_ = wide.find_first_not_of(static_cast<wchar_t>('\\'), first - wide.begin());
-                    auto after = (after_ == std::string::npos ? wide.end() : wide.begin() + after_);
-
-                    if (after == wide.end() || *after == static_cast<wchar_t>('"')) {
-                        /* Duplicate (escape) all backslashes before (possibly ending) quote. */
-                        arguments.insert(arguments.end(), first, after);
-                        arguments.insert(arguments.end(), first, after);
-                    } else {
-                        /* Not before quote, no need to escape backslashes. */
-                        arguments.insert(arguments.end(), first, after);
-                    }
-
-                    /* Jump to the next non-backslash character. */
-                    it = std::prev(after);
-                } else {
-                    abort();
-                }
-            }
+    WideString executablePath = StringToWideString(context->executablePath());
 
-            arguments += static_cast<wchar_t>('"');
-        }
+    WideString arguments = EscapedToken(executablePath);
+    for (std::string const &argument : context->commandLineArguments()) {
+        arguments += static_cast<wchar_t>(' ');
```

Comment:
I think you want to do
```
arguments.push_back(L' ');
```
instead of that `static_cast`. The same applies for all the wchar_t you're trying to create.

Also, *technically*, the character type for `WideString` is `WCHAR`, not `wchar_t`.